src/main/java/com/datapipe/jenkins/vault/model/VaultSecretValue.java

```diff
@@ -39,25 +39,35 @@
 public class VaultSecretValue
     extends AbstractDescribableImpl<VaultSecretValue> {
 
+    public final static Boolean DEFAULT_IS_REQUIRED = true;
+
     private String envVar;
+    private boolean isRequired;
     private final String vaultKey;
 
     @Deprecated
     public VaultSecretValue(String envVar, @NonNull String vaultKey) {
         this.envVar = fixEmptyAndTrim(envVar);
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = DEFAULT_IS_REQUIRED;
     }
 
     @DataBoundConstructor
     public VaultSecretValue(@NonNull String vaultKey) {
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = DEFAULT_IS_REQUIRED;
```

Comment:
This would still force to always default to true.

The correct behavior is to set the default on the .jelly file :)
https://github.com/jenkinsci/compuware-topaz-for-total-test-plugin/blob/e3c8508809e1befad31e6908785322a06fbbe944/src/main/resources/com/compuware/jenkins/totaltest/TotalTestCTBuilder/config.jelly#L65

Reply:
A couple quick questions on this... Does this mean I shouldn't have the lines
```java
this.isRequired = DescriptorImpl.DEFAULT_IS_REQUIRED;
```
at all? Or I should just have those **and** the jelly file changes?

Additionally, just for my own curiosity, how does all of this work when calling the vault plugin from a declarative pipeline as opposed to using the global GUI config (which I assume is what the .jelly files are for)?

Reply:
This should not be there at all in the constructor.

The default should be added to the private field definition: see https://github.com/jenkinsci/compuware-topaz-for-total-test-plugin/blob/e3c8508809e1befad31e6908785322a06fbbe944/src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java#L136